packages/yswaps/scripts/libraries/utils/should-execute-trade.ts

```diff
@@ -0,0 +1,28 @@
+import { ethers } from 'hardhat';
+import { wallet } from '../../../../commons/test-utils';
+import { IERC20, IERC20__factory } from '../../../typechained';
+import { SimpleEnabledTrade } from '../types';
+
+export async function shouldExecuteTrade({
+  strategy,
+  trades,
+  checkType,
+}: {
+  strategy: string;
+  trades: SimpleEnabledTrade[];
+  checkType: 'total' | 'partial';
+}): Promise<boolean> {
+  const executeConfirmations: boolean[] = [];
+  const randomAddress = await wallet.generateRandomAddress();
+  for (const trade of trades) {
+    const token = await ethers.getContractAt<IERC20>(IERC20__factory.abi, trade.tokenIn, randomAddress);
```

Comment:
```suggestion
  for (const trade of trades) {
    const token = await ethers.getContractAt<IERC20>(IERC20__factory.abi, trade.tokenIn);
```
There should be no need of setting the signer if you are using `getContractAt` is just needed when you do `IERC20__factory.connect(address, signer)`